src/server/streamable-http.ts

```diff
@@ -0,0 +1,388 @@
+        },
+        id: null
+      }));
+    }
+  }
+
+  /**
+   * Handles GET requests to establish SSE connections
+   */
+  private async handleGetRequest(req: IncomingMessage, res: ServerResponse): Promise<void> {
+    // validate the Accept header
+    const acceptHeader = req.headers.accept;
+    if (!acceptHeader || !acceptHeader.includes("text/event-stream")) {
+      res.writeHead(406).end(JSON.stringify({
+        jsonrpc: "2.0",
+        error: {
+          code: -32000,
+          message: "Not Acceptable: Client must accept text/event-stream"
+        },
+        id: null
+      }));
+      return;
+    }
+
+    const connectionId = randomUUID();
+    const lastEventId = req.headers["last-event-id"];
+    const lastEventIdStr = Array.isArray(lastEventId) ? lastEventId[0] : lastEventId;
+
+    res.writeHead(200, {
+      "Content-Type": "text/event-stream",
+      "Cache-Control": "no-cache",
+      Connection: "keep-alive",
+      "mcp-session-id": this._sessionId,
+    });
+
+    const connection: StreamConnection = {
+      response: res,
+      lastEventId: lastEventIdStr,
+      messages: [],
+    };
+
+    this._connections.set(connectionId, connection);
+
+    // if there is a Last-Event-ID, replay messages on this connection
+    if (lastEventIdStr) {
+      this.replayMessages(connectionId, lastEventIdStr);
+    }
+
+    res.on("close", () => {
+      this._connections.delete(connectionId);
+      // remove all request mappings associated with this connection
+      for (const [reqId, connId] of this._requestConnections.entries()) {
+        if (connId === connectionId) {
+          this._requestConnections.delete(reqId);
+        }
+      }
+      if (this._connections.size === 0) {
+        this.onclose?.();
+      }
+    });
+  }
+
+  /**
+   * Handles POST requests containing JSON-RPC messages
+   */
+  private async handlePostRequest(req: IncomingMessage, res: ServerResponse): Promise<void> {
+    try {
+      // validate the Accept header
+      const acceptHeader = req.headers.accept;
+      if (!acceptHeader || 
+         (!acceptHeader.includes("application/json") && !acceptHeader.includes("text/event-stream"))) {
+        res.writeHead(406).end(JSON.stringify({
+          jsonrpc: "2.0",
+          error: {
+            code: -32000,
+            message: "Not Acceptable: Client must accept application/json and/or text/event-stream"
+          },
+          id: null
+        }));
+        return;
+      }
+
+      const ct = req.headers["content-type"];
+      if (!ct || !ct.includes("application/json")) {
+        res.writeHead(415).end(JSON.stringify({
+          jsonrpc: "2.0",
+          error: {
+            code: -32000,
+            message: "Unsupported Media Type: Content-Type must be application/json"
+          },
+          id: null
+        }));
+        return;
+      }
+
+      const parsedCt = contentType.parse(ct);
+      const body = await getRawBody(req, {
```

Comment:
Shall we consider accepting parsed request body here just like: https://github.com/modelcontextprotocol/typescript-sdk/blob/main/src/server/sse.ts#L68

In some cases the MCP Server lives in another application server, and that application server may have global body parsers in place.

Reply:
Now it should be able work like:

```ts
app.post('/mcp', (req, res) => {
   transport.handleRequest(req, res, req.body);
});
```